crates/sui-replay/src/displays/transaction_displays.rs

```diff
@@ -179,8 +187,84 @@ impl<'a> Display for Pretty<'a, Argument> {
             Argument::GasCoin => "GasCoin".to_string(),
             Argument::Input(i) => format!("Input  {}", i),
             Argument::Result(i) => format!("Result {}", i),
-            Argument::NestedResult(j, k) => format!("Nested Result {}: {}", j, k),
+            Argument::NestedResult(j, k) => format!("Result {}: {}", j, k),
         };
         write!(f, "{}", output)
     }
 }
+
+impl<'a> Display for Pretty<'a, ExecutionResult> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let Pretty((mutable_ref_outputs, return_values)) = self;
+        let len_m_ref = mutable_ref_outputs.len();
+        let len_ret_vals = return_values.len();
+        if len_ret_vals > 0 || len_m_ref > 0 {
+            write!(f, "\n ┌ ")?;
+        }
+        if len_m_ref > 0 {
+            write!(f, "\n │ Mutable Reference Outputs:")?;
+        }
+        for (arg, bytes, type_tag) in mutable_ref_outputs {
+            write!(f, "\n │   {} ", arg)?;
+            resolve_value(f, bytes, type_tag)?;
+        }
+        if len_ret_vals > 0 {
+            write!(f, "\n │ Return Values:")?;
+        }
+
+        for (i, (bytes, type_tag)) in return_values.iter().enumerate() {
+            write!(f, "\n │   {i:<2} ")?;
+            resolve_value(f, bytes, type_tag)?;
+        }
+        if len_ret_vals > 0 || len_m_ref > 0 {
+            write!(f, "\n └")?;
+        }
+        Ok(())
+    }
+}
+
+impl<'a> Display for Pretty<'a, TypeTag> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let Pretty(type_tag) = self;
+        match type_tag {
+            TypeTag::Vector(v) => {
+                write!(f, "Vector of {}", Pretty(&**v))
+            }
+            TypeTag::Struct(s) => {
+                write!(f, "{}::{}", s.module, s.name)
+            }
+            _ => {
+                write!(f, "{}", type_tag)
+            }
+        }
+    }
+}
+
+fn resolve_value(f: &mut Formatter<'_>, bytes: &[u8], type_tag: &TypeTag) -> std::fmt::Result {
+    let value = SuiJsonValue::from_bcs_bytes(None, bytes)
```

Comment:
@tzakian @tnowacki here the bytes and type tag are getting resolved into the actual value as suggested in a comment. We don't display the array and object values because the data is way to large and unpredictable to be formatted in a readable and or useful way via CLI, but someone can hopefully find the comment I left and update the code if they really want to see it.

Reply:
I think it would be super helpful to be able to see object values, since that is likely where we would need to introspect in an emergency situation. 

If we get a VM instance, we can use
external-crates/move/crates/move-core-types/src/annotated_value.rs
And improve any rendering there overtime. @tzakian might have some ideas there

Reply:
Ah looks like @tzakian had the same thought and mentioned annotated values in his other comment :)

Reply:
Yea, I think annotated value is what we want here. Moreover, annotated values have a pretty nice display already built in to them that you might be able to use. You can get a sense for what the formatted values might look like here: https://github.com/MystenLabs/sui/blob/main/external-crates/move/crates/move-cli/tests/sandbox_tests/print_values/args.exp#L67